lib/param_parsing/terragrunt.go

```diff
@@ -0,0 +1,45 @@
+package param_parsing
+
+import (
+	"github.com/hashicorp/hcl2/gohcl"
+	"github.com/hashicorp/hcl2/hclparse"
+	"github.com/warrensbox/terraform-switcher/lib"
+	"os"
+)
+
+const terraGruntFileName = "terragrunt.hcl"
+
+type terragruntVersionConstraints struct {
+	TerraformVersionConstraint string `hcl:"terraform_version_constraint"`
+}
+
+func GetVersionFromTerragrunt(params Params) Params {
+	filePath := params.ChDirPath + "/" + terraGruntFileName
+	if lib.CheckFileExist(filePath) {
+		logger.Infof("Reading configuration from %q", filePath)
+		parser := hclparse.NewParser()
+		hclFile, diagnostics := parser.ParseHCLFile(filePath)
+		if diagnostics.HasErrors() {
+			logger.Fatalf("Unable to parse HCL file %q", filePath)
+			os.Exit(1)
```

Comment:
As of addition of `logger.ExitFunc = os.Exit` to this PR, is `os.Exist(1)` still needed? 🤔